pkg/aws/apis/validation/validation_test.go

```diff
@@ -1334,6 +1334,99 @@ var _ = Describe("Validation", func() {
 					},
 				},
 			}),
+			Entry("AWS machine class with instanceMetadata", &data{
+				setup: setup{},
+				action: action{
+					spec: &awsapi.AWSProviderSpec{
+						AMI: "ami-123456789",
+						BlockDevices: []awsapi.AWSBlockDeviceMappingSpec{
+							{
+								Ebs: awsapi.AWSEbsBlockDeviceSpec{
+									VolumeSize: 50,
+									VolumeType: "gp3",
+									Iops:       3500,
+									Throughput: aws.Int64(200),
+								},
+							},
+						},
+						IAM: awsapi.AWSIAMProfileSpec{
+							Name: "test-iam",
+						},
+						Region:      "eu-west-1",
+						MachineType: "m4.large",
+						KeyName:     pointer.String("test-ssh-publickey"),
+						NetworkInterfaces: []awsapi.AWSNetworkInterfaceSpec{
+							{
+								SecurityGroupIDs: []string{
+									"sg-00002132323",
+								},
+								SubnetID: "subnet-123456",
+							},
+						},
+						Tags: map[string]string{
+							"kubernetes.io/cluster/shoot--test": "1",
+							"kubernetes.io/role/test":           "1",
+						},
+						InstanceMetadataOptions: &awsapi.InstanceMetadataOptions{
+							HTTPPutResponseHopLimit: pointer.Int64(32),
+						},
+					},
+					secret: providerSecret,
+				},
+				expect: expect{
+					errToHaveOccurred: false,
+				},
+			}),
+			Entry("AWS machine class with invalid instanceMetadata.httpPutResponseHopLimit.", &data{
+				setup: setup{},
+				action: action{
+					spec: &awsapi.AWSProviderSpec{
+						AMI: "ami-123456789",
+						BlockDevices: []awsapi.AWSBlockDeviceMappingSpec{
+							{
+								Ebs: awsapi.AWSEbsBlockDeviceSpec{
+									VolumeSize: 50,
+									VolumeType: "gp3",
+									Iops:       3500,
+									Throughput: aws.Int64(200),
+								},
+							},
+						},
+						IAM: awsapi.AWSIAMProfileSpec{
+							Name: "test-iam",
+						},
+						Region:      "eu-west-1",
+						MachineType: "m4.large",
+						KeyName:     pointer.String("test-ssh-publickey"),
+						NetworkInterfaces: []awsapi.AWSNetworkInterfaceSpec{
+							{
+								SecurityGroupIDs: []string{
+									"sg-00002132323",
+								},
+								SubnetID: "subnet-123456",
+							},
+						},
+						Tags: map[string]string{
+							"kubernetes.io/cluster/shoot--test": "1",
+							"kubernetes.io/role/test":           "1",
+						},
+						InstanceMetadataOptions: &awsapi.InstanceMetadataOptions{
+							HTTPPutResponseHopLimit: pointer.Int64(72),
+						},
+					},
+					secret: providerSecret,
+				},
+				expect: expect{
+					errToHaveOccurred: true,
+					errList: field.ErrorList{
+						{
+							Type:     "FieldValueInvalid",
+							Field:    "providerSpec.instanceMetadata.httpPutResponseHopLimit",
+							BadValue: int64(72),
+							Detail:   "Only values between 0 and 64, both included, are accepted",
+						},
+					}},
```

Comment:
You have only written tests for hop limits, however validations have also been added for httpendpoint. Can you add those tests as well?

Reply:
Added with latest commit